openedx/features/course_experience/course_tools.py

```diff
@@ -0,0 +1,52 @@
+"""
+Support for course tool plugins.
+"""
+from openedx.core.lib.api.plugins import PluginManager
+
+# Stevedore extension point namespace
+COURSE_TOOLS_NAMESPACE = 'openedx.course_tool'
+
+
+class CourseTool(object):
+    """
+    This is an optional base class for Course Tool plugins.
+
+    Plugin implementations can choose to subclass CourseTool to get
+    useful default behavior, but it is not a requirement. A course tool
+    plugin should define the following class methods:
+
+     * is_enabled(cls, course_key):
+       Returns true if this tool is enabled for the specified course key.
+     * title(cls):
+       Returns the title of this tool.
+     * icon_classes(cls):
+       Returns the icon classes needed to represent this tool.
+     * url(cls, course_key):
+       Returns the URL for this tool for the specified course key.
+    """
```

Comment:
I "think" I learned it from @nedbat, but I could be wrong.  I remember it being around getting more accurate diffs when adding more docs, but I could be entirely making this up. :)